src/core/service.c

```diff
@@ -1358,10 +1372,69 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        struct stat st;
+                        _cleanup_close_ int fd = -1;
+
+                        if (stat(f->path, &st) < 0)
+                                return log_error_errno(errno, "Failed to stat %s: %m", f->path);
+
+                        if (S_ISSOCK(st.st_mode)) {
+                                struct sockaddr_un addr;
```

Comment:
please use:

```c
union sockaddr_union addr;
```

instead. that way you can avoid the ugly casting during connect()